### DISCUSSION TITLE
Build bundle on top of another bundle

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#352] [OPA and Rego] [answered]
Is it possible to build a bundle file from current directory and combining with another bundle file?

**Comment by anderseknert** [comment, marked as answer]
Yes. You'll need to ensure you provide a `.manifest` file in the bundle that denotes the roots of the bundle, or else you'll see conflicts like: 

```shell
error: bundle merge failed: manifest has overlapped roots: '' and ''
```
Provided that you have a directory structure similar to:
```shell
$ tree -a
.
├── bundle1
│   ├── .manifest
│   ├── p1
│   │   └── data.json
│   └── p1.rego
└── bundle2
    ├── .manifest
    ├── p2
    │   └── data.json
    └── p2.rego

5 directories, 6 files
```
You can build the first bundle:
```shell
$ opa build -o bundle1.tar.gz -b bundle1
```
And then a second bundle using both the `bundle2` directory and the first bundle as input:
```shell
opa build -o bundle2.tar.gz bundle2 -b bundle1
```

There seems to be a bug in that unless you explicitly provide a `data.json` file in each bundle directory, the result of merging the two bundles will be a `data.json` file containing only `null`. I'll file an issue for that.

With all that said, I would prefer to build a single bundle from directory structure only. So another, and in my opinion better, option would be to first unpack the other bundle, and then build a new one from the contents of that and your other directory.

**Reply by Ronnie-personal** [reply]
Thanks for providing the details, it's very helpful!
